pkg/cache/cluster.go

```diff
@@ -554,16 +554,19 @@ func (c *clusterCache) watchEvents(ctx context.Context, api kube.APIResourceInfo
 				if err != nil {
 					return fmt.Errorf("failed to load initial state of resource %s: %v", api.GroupKind.String(), err)
 				}
-
-				return runSynced(&c.lock, func() error {
-					c.replaceResourceCache(api.GroupKind, items, ns)
-					return nil
-				})
+				return nil
 			})
 
 			if err != nil {
 				return err
 			}
+
+			if err := runSynced(&c.lock, func() error {
+				c.replaceResourceCache(api.GroupKind, items, ns)
+				return nil
+			}); err != nil {
+				return err
+			}
```

Comment:
Because the callback func is not possible to return nil, the `runSynced` is not really helpful and so I think this will be clearer to be written:

```golang
c.lock.Lock()
c.replaceResourceCache(api.GroupKind, items, ns)
c.lock.Release()
```

Reply:
Good point! Updated code.